postgres/postgres.go

```diff
@@ -56,7 +56,44 @@ func (ds Datastore) AddFilter(host string, newFilter *models.Filter) (models.Fil
 	return *newFilter, nil
 }
 
-// AddDimensions method adds dimensions to be stored in postgres and relates them to a filter job
+// UpdateFilter updates a given filter against a given dataset in the postgres databaseilter job
+func (ds Datastore) UpdateFilter(host string, filter *models.Filter) error {
+	tx, err := ds.db.Begin()
+	if err != nil {
+		return err
+	}
+
+	currentFilterJob, err := getFilterJob(tx, ds, filter.FilterID)
+	if err != nil {
+		log.Error(err, log.Data{"filter_job_id": filter.FilterID, "current_filter_job": currentFilterJob})
+		return err
+	}
+
+	updateStatement, err := updateStatement(filter)
+	if err != nil {
+		return err
+	}
+
+	stmt, err := tx.Prepare(updateStatement)
+	if err != nil {
+		return err
+	}
+
+	_, err = stmt.Exec(filter.FilterID)
+	if err != nil {
+		return err
```

Comment:
So no as I haven't been able to change anything against the database.

The statement previously only retrieves the data to check the state before doing an update to it. So to stop another update or delete to happen against this data (filter job) I have placed both sql statements in a transaction to lock the row, this will prevent any changes occurring between the two queries/statements.